### PR TITLE
Cherry-pick RTNR commits from main branch to adl-003-drop-stable

### PR DESCRIPTION
**Pull request by MingJenTai**
This PR cherry-picks RTNR commits from main branch to adl-003-drop-stable.
The original commits and modifications are stated in the commit message.

### CODE REVIEW

**Comment by cujomalainey** [comment]
Thanks for the update, FYI, you can add the cherry picked from tag with the `-x` flag when doing the cherry pick

**Review by cujomalainey — APPROVED** [review]

**Review by mwasko — APPROVED** [review]

File: tools/topology/sof-tgl-max98357a-rt5682.m4 [hunk 1]
```diff
@@ -97,8 +97,18 @@ define(DMIC_PIPELINE_KWD_ID, `12')
...
+ifdef(`RTNR',
+`define(DMICPROC, rtnr)',
+`ifdef(`DMICPROC', , `define(DMICPROC, eq-iir-volume)')')
+
+ifdef(`RTNR',
+`define(DMIC16KPROC, rtnr)',
+`ifdef(`DMIC16KPROC', , `define(DMIC16KPROC, eq-iir-volume)')')
+
+# include the generic dmic if RTNR is defined, else include generic dmic with kwd
+include(ifdef(`RTNR', platform/intel/intel-generic-dmic.m4, platform/intel/intel-generic-dmic-kwd.m4))
```
Comment (cujomalainey) [inline tools/topology/sof-tgl-max98357a-rt5682.m4:111]: i missed this in my initial review, @MingJenTai this wont work. You simply can't remove hotwording. They both need to work in this config.

Comment (MingJenTai) [reply tools/topology/sof-tgl-max98357a-rt5682.m4:111]: @cujomalainey Ok I'll have a fix for that. Thanks.

Comment (MingJenTai) [reply tools/topology/sof-tgl-max98357a-rt5682.m4:111]: @cujomalainey I've created PR https://github.com/thesofproject/sof/pull/4988 for this. Thank you.

**Pull request merged**